wordpress-mobile/WordPress-Android
# Add block editor tests

### Description
After the classic editor [was made unavailable for new posts](https://github.com/wordpress-mobile/WordPress-Android/pull/14381) a couple of [tests covering post creation were disabled](https://github.com/wordpress-mobile/WordPress-Android/pull/14381/files#).
Since the ability to create a post is considered critical to be automated, this PR adds a few tests to the new block editor covering the disabled scenarios and a new scenario covering the `HTML mode`.


- `publishSimplePost`: checks that a simple post with only title and text can be published.
- `publishFullPost`: checks that a post with title, text, image, category and tag can be published.
- `blockEditorCanDisplayElementAddedInHtmlMode`: checks that an element aded in HTML Mode can be displayed in Visual mode.

### Testing Instructions
- The[ Connected Tests-2 CI job is green](https://app.circleci.com/pipelines/github/wordpress-mobile/WordPress-Android/28300/workflows/d70ce190-93db-4d98-ad7f-c2252beb7a34/jobs/117946/steps), and you can see `publishSimplePost`,`publishFullPost` and blockEditorCanDisplayElementAddedInHtmlMode tests included.
- Associated [Firebase execution](https://console.firebase.google.com/u/0/project/api-project-108380595987/testlab/histories/bh.e0c3a59bd9ed670/matrices/5928684839087902793/executions/bs.f89d7e4e5549d8d9/test-cases) is also green (Firebase console access is needed to check this).
- Tests run locally on Pixel 2 API 28 (this is the model used on CI).

## Regression Notes
1. Potential unintended areas of impact
Changes would only affect ui tests.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Checked that all tests are passing locally an on CI.

3. What automated tests I added (or what prevented me from doing so)
- `publishSimplePost`
- `publishFullPost`
- `blockEditorCanDisplayElementAddedInHtmlMode`

PR submission checklist:
